### PR TITLE
Add feature to create callbacks and validate keypairs

### PR DESCRIPTION
**Pull request by DhruvKhemani**
This new feature allows the user to add callback functions to the insertion/removal of a smartcard. This is an essential and long time requested feature that finally works from within the library.
Usage of this new feature will be shown in an upcoming example!

### CODE REVIEW

**Comment by DhruvKhemani** [comment]
Added a new feature that check if a keypair is actually valid/even exists... This is a very simple yet useful command.

**Comment by DhruvKhemani** [comment]
As of right now an example for the `is_key_valid(reader, key_id)` command won't be done separately, instead, it was incorporated into generate and validate signature example (PR #17)

File: blocksec2go/commands.py [hunk 1]
```diff
@@ -146,6 +162,30 @@ def get_key_info(reader, key_id):
...
+    
+    Raises:
+        RuntimeError: Entered key_id is outside of keypair scope.
+        
+        Any exceptions thrown by the reader wrapper are passed through.
+    """
+    logger.debug('IS KEY VALID key %d', key_id)
+    if key_id < 0 or key_id > 255:
+        raise RuntimeError('Invalid key_id: ' + str(key_id))
+    global_counter, counter, key = get_key_info(reader, key_id)
```
Comment (ahoeller) [inline blocksec2go/commands.py:184]: Pls. also catch CardError exception that could be raised by get_key_info if key_id is valid but there was an error during key generation

**Review by ahoeller — CHANGES_REQUESTED** [review]
Please add the small addition in the verify_key function

File: blocksec2go/commands.py (hunk 1 above)
Comment (DhruvKhemani) [reply blocksec2go/commands.py:184]: This was done on purpose so that the user can catch it...

**Comment by DhruvKhemani** [comment]
Done

**Pull request merged**